source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,214 @@
+
+    """
+    height = np.shape(rgb)[0]
+    width = np.shape(rgb)[1]
+    pixel_hight_start_index = int((height - pixels) / 2)
+    pixel_hight_end_index = int((height + pixels) / 2)
+    pixel_width_start_index = int((width - pixels) / 2)
+    pixel_width_end_index = int((width + pixels) / 2)
+    red = rgb[
+        pixel_hight_start_index:pixel_hight_end_index,
+        pixel_width_start_index:pixel_width_end_index,
+        0,
+    ]
+    green = rgb[
+        pixel_hight_start_index:pixel_hight_end_index,
+        pixel_width_start_index:pixel_width_end_index,
+        1,
+    ]
+    blue = rgb[
+        pixel_hight_start_index:pixel_hight_end_index,
+        pixel_width_start_index:pixel_width_end_index,
+        2,
+    ]
+    mean_red = np.mean(np.reshape(red, -1), dtype=np.float64)
+    mean_green = np.mean(np.reshape(green, -1), dtype=np.float64)
+    mean_blue = np.mean(np.reshape(blue, -1), dtype=np.float64)
+
+    return mean_red, mean_green, mean_blue
+
+
+def _apply_color_balance(rgb, red_balance, blue_balance):
+    """Apply color balance to RGB image.
+
+    Args:
+        rgb: Input RGB image
+        red_balance: Red balance
+        blue_balance: Blue balance
+
+    Returns:
+        corrected_image: RGB image after color balance
+
+    """
+    default_red_balance = zivid.Settings().red_balance
+    default_blue_balance = zivid.Settings().blue_balance
+    corrected_rgb = np.copy(rgb)
+    corrected_rgb[:, :, 0] = rgb[:, :, 0] * red_balance / default_red_balance
+    corrected_rgb[:, :, 2] = rgb[:, :, 2] * blue_balance / default_blue_balance
+
+    return corrected_rgb
+
+
+def _color_balance_calibration(camera, settings_3d):
+    """Balance color for RGB image by taking images of white surface (piece of paper, wall, etc.) in a loop.
+
+    Args:
+        camera: Zivid camera
+        settings_3d: 3D capture settings
+
+    Returns:
+        corrected_red_balance: Corrected red balance
+        corrected_blue_balance: Corrected blue balance
+
+    """
+    print(f"Starting color balance calibration")
+    corrected_red_balance = 1.0
+    corrected_blue_balance = 1.0
+    settings_list = []
+    settings_list.append(settings_3d)
+    first_iteration = True
+    while True:
+        settings_3d.red_balance = corrected_red_balance
+        settings_3d.blue_balance = corrected_blue_balance
+        frame = zivid.hdr.capture(camera, settings_list)
+        point_cloud = frame.get_point_cloud().to_array()
+        rgb = np.dstack([point_cloud["r"], point_cloud["g"], point_cloud["b"]])
+        if first_iteration:
+            _display_rgb(rgb, "RGB image before color balance (3D capture)")
+            first_iteration = False
+        [mean_red, mean_green, mean_blue] = _compute_mean_rgb(rgb, 100)
+        print(
+            (
+                "Mean color values: R = "
+                f"{int(mean_red)} "
+                "G = "
+                f"{int(mean_green)} "
+                f"B = "
```

Comment:
Minor: This does not need to be an f-string

Reply:
Done.